source/python-best-practise.rst

```diff
@@ -0,0 +1,498 @@
+    license = MIT
+    license_file = LICENSE
+    project_urls =
+        Documentation = https://Kortforsyningen.github.io/...
+        Source = https://github.com/Kortforsyningen/...
+        Tracker = https://github.com/.../issues
+
+    [options]
+    zip_safe = False
+    package_dir =
+        = src
+    packages = find:
+    platforms = any
+    python_requires = >=3.10
+
+Med ovenstående opsætning kan Pythons pakke-styringsværkøj ``pip`` selv finde ud af at installere pakken ``setuptools``, som bygger pakken med de givne metadata, som ``pip`` så installerer.
+
+For at installere pakken, så den er tilgængelig for Python-fortolkeren, vi bruger i conda-miljøet ``package-dev``, bruger vi ``pip`` som et modul i det aktiverede miljø, så vi er sikre på, at vi ikke bruger en anden ``pip``-kommando, der kan være tilgængelig i terminalen:
+
+.. code-block :: none
+
+    (package-dev)> python -m pip install -e .
+
+Læs mere om de enkelte konfigurationsmuligheder i dokumentationen for `SetupTools`_.
+
+.. _`SetupTools`: https://setuptools.pypa.io/
+
+
+Test-funktionalitet
+-------------------
+
+Denne vejledning har et separat kapitel om at tests og anden kvalitetssikring i Python. Her nævner vi kort, at al test-funktionalitet bør ligge separat i sin egen mappe kaldet ``tests/``.
+
+Den rene Python-pakke med test-funktionaliteten ser således ud:
+
+.. code-block:: none
+
+    package
+    ├── environment-dev.yml
+    ├── LICENSE
+    ├── pyproject.toml
+    ├── README.md
+    ├── setup.cfg
+    ├── src
+    │   └── package
+    │       ├── module1.py
+    │       ├── module2.py
+    │       └── __init__.py
+    └── tests
+        └── package
+            ├── test_module1.py
+            └── test_module2.py
+
+Bemærk, at test-koden følger samme struktur som pakkens undermapper.
+
+
+
+
+Brugerens synsvinkel: installation og dokumentation
+---------------------------------------------------
+
+For brugeren er kun produktet og den brugervendte dokumentation relevant.
+
+Her kan brugeren installere pakken ved at klone projekt-mappen ned med Git. Her skal brugeren først checke koden ud og dernæst manuelt oprette et miljø og installere de pakker (Afhængigheder), som vores program skal bruge. Python er forudsat installeret hos brugeren, og det er antaget, at brugeren kan bruge det.
+
+Man kan i ovenstående tilfælde distribuere koden til et pakke-arkiv som the Python Package Index (PyPI). For brugeren ville det derfor være væsentligt lettere at installere pakken i et arbitrært mamba-miljø.
+
+Der kan være flere grunde til, at vi ikke distribuerer koden til et (globalt) Python-pakke-arkiv. Én årsag kan være, at vi kan have brug for, at brugeren tester en specifik version af koden, hvilket er nemt, hvis brugeren bare skal checke den givne version ud kortvarigt.
+
+
+
+Dokumentation
+=============
+
+
+[...]
+
+
+
+Typing annotations
+==================
+
+[...]
+
+Skal dete afsnit over i vætkrøjer, hvor vi kan tale om ``mypy``?
```

Comment:
Spørgsmålet er om det ikke kan give mening med en dedikeret side til god kodeskik. Her kan PEP8, zen of python, type annotations, diverser idiomer (exceptions, iterators, context managers, ...) og andet der hjælper os med at gøre koden pythonsk i sit udtryk.

Reply:
Jo, jeg synes også, at det er en stor nok mundfuld til, at det basale med pakke-opbygningen vil stå klarere som sådan (pakkehåndtering), hvis det er for sig selv.

Men så skal vi nok ændre på dokumentets filnavn og overskrift, så det er klart, at der kun er tale om projekt-mappestruktur og pakke-opsætning her.